coalib/io/Directory.py

```diff
@@ -0,0 +1,89 @@
+import os
+
+from coalib.parsing.Globbing import relative_recursive_glob
+from coala_utils.decorators import generate_eq
+
+
+@generate_eq('path', 'timestamp')
+class Directory:
+    """
+    The `Directory` class acts as an interface to directories and
+    provides useful information about them such as:
+
+        * The directory path.
+        * The parent directory path.
+        * The last modified timestamp of the directory.
+        * The number of children contained inside a directory.
+
+    >>> import os
+    >>> d = Directory('tests/io/DirectoryTestDir/')
```

Comment:
Again temporary files/directories? ;)

Reply:
I wanted to show the different outputs given by the `get_children` and `get_children_recursively` methods.

I'm not sure if we can set up directories and files inside temporary  directories since they are all random. Even if we can do that using `tempfile` I would have to create more at least more than 2 temp files/dirs, I'm not sure if creating so many tempfiles or dirs would look good in the docstrings .
Using an already existing directory structure lets us avoid that and show that difference between different methods too.

Still if you want me to change that let me know.

Reply:
You can create a top-level directory, and then create inside the structure you need. But that's definitely more cumbersome. I'm also not sure if we should support executing pytest from different directories^^ Ideally yes, better than not, though makes things certainly harder.